packages/node/src/utils/promise-utils.ts

```diff
@@ -25,22 +24,15 @@ export function goTimeout<T>(ms: number, fn: Promise<T>): Promise<Response<T>> {
   return go(promiseTimeout(ms, fn));
 }
 
-// A native implementation of the following function might look like:
-//
-//   function promiseTimeout<T>(ms: number, promise: Promise<T>): Promise<T> {
-//     const timeout = new Promise((_res, reject) => {
-//       setTimeout(() => {
-//         reject(new Error(`Timed out in ${ms} ms.`));
-//       }, ms);
-//     });
-//     return Promise.race([promise, timeout]);
-//   }
-//
-// The problem with this is that that the slow promise still runs until it resolves.
-// This means that the serverless function will not exit until the entire timeout
-// duration has been reached which is a problem.
 export function promiseTimeout<T>(ms: number, promise: Promise<T>): Promise<T> {
-  return Bluebird.resolve(promise).timeout(ms);
+  const timeout = new Promise((_res, reject) => {
+    const id = setTimeout(() => {
+      clearTimeout(id);
```

Comment:
`clearTimeout` is useless here.

You want to call it other way around -- when the original promise resolves/rejects.
(Maybe that's the reason you need the process.exit calls?)

Reply:
Thanks, how about this?

```ts
let timeoutId: NodeJS.Timeout;
const timeout = new Promise((_res, reject) => {
  timeoutId = setTimeout(() => {
    reject(new Error(`Operation timed out in ${ms} ms.`));
  }, ms);
});

const wrappedPromise = promise.finally(() => {
  if (timeoutId) {
    clearTimeout(timeoutId);
  }
});

return Promise.race([wrappedPromise, timeout]) as Promise<T>;
```

Reply:
Yep, lgtm. Did it fix the issue with explicit process.exit call?

Reply:
It did actually fix it 👍🏻